uint16 distance_1
uint16 distance_2
uint16 distance_3
uint16 distance_4
